src/core/contract_address/casm_contract_address.rs

```diff
@@ -0,0 +1,190 @@
+use crate::core::errors::contract_address_errors::ContractAddressError;
+use crate::services::api::contract_classes::deprecated_contract_class::EntryPointType;
+use cairo_lang_starknet::casm_contract_class::{CasmContractClass, CasmContractEntryPoint};
+use cairo_vm::felt::Felt252;
+use starknet_crypto::{poseidon_hash_many, FieldElement};
+
+const CONTRACT_CLASS_VERSION: &[u8] = b"COMPILED_CLASS_V1";
+
+fn get_contract_entry_points_hashed(
+    contract_class: &CasmContractClass,
+    entry_point_type: &EntryPointType,
+) -> Result<FieldElement, ContractAddressError> {
+    let contract_entry_points = get_contract_entry_points(contract_class, entry_point_type);
+
+    // for each entry_point, we need to store 3 FieldElements: [selector, offset, poseidon_hash_many(builtin_list)].
+    let mut entry_points_flatted = Vec::with_capacity(contract_entry_points.len() * 3);
+
+    for entry_point in contract_entry_points {
+        entry_points_flatted.push(
+            // fix this conversion later
+            FieldElement::from_bytes_be(&Felt252::from(entry_point.selector).to_be_bytes())
+                .map_err(|_err| {
+                    ContractAddressError::Cast("Felt252".to_string(), "FieldElement".to_string())
+                })?,
+        );
+        entry_points_flatted.push(FieldElement::from(entry_point.offset));
+        let mut builtins_flatted = Vec::new();
+        entry_point.builtins.iter().for_each(|builtin| {
+            builtins_flatted.push(FieldElement::from_byte_slice_be(builtin.as_bytes()).unwrap());
+        });
```

Comment:
We could use `map` + `collect` here.

Reply:
Yes, you're right. Also you made me realize that I forgot to remove an unwrap 😓 ("Contains unwrap" octagon needed).

Reply:
<img src="https://github.com/lambdaclass/starknet_in_rust/assets/47506558/e55a7ca6-cad9-4d93-b78d-8477b882d9e7" width="200" height="200" /> ?